django/apps/registry.py

```diff
@@ -331,13 +331,19 @@ def unset_installed_apps(self):
         self.apps_ready = self.models_ready = self.ready = True
         self.clear_cache()
 
-    def clear_cache(self):
+    def clear_cache(self, expire_models_cache=False):
```

Comment:
The boolean was introduced because you can't expire the model cache in *all* cases - the apps registry needs to be ready before you can flush the model cache, because you have to interrogate the list of models to build the model cache. The apps registry does aggressive self-flushing every time a model is loaded, so this causes a problem.

However, if you just don't flush the model cache util "apps.ready", you can avoid this problem. This stands to reason AFAICT; you won't be doing any model operations until the app cache is populated, and the model cache is populated on first use, so there can't be a model cache until the apps cache is ready.